### ISSUE TITLE
Add an ability to call a PTZ preset point

### ISSUE THREAD

**Issue by Brix1330** [user2684/imou_life#75]
I would love to get this feature implemented, especially for Cruiser 4MP, so people can automate the camera's pan-and-tilt location to a saved preset. IIRC currently it's only possible to move the camera manually from HA.
 
I've integrated my camera with Frigate to detect humans at my gate. The default view covers both part of the main road and the gate. However, with this approach, I can only set the camera to a fixed view as automatic tracking would interfere with the Frigate detection area and with the automatic tracking turned on, someone could trick the camera to move away from the POI. 
If I could call a preset, when there's a human at my gate, it would switch to a preset point (number dropdown menu; just like on SmartPSS) that points towards the gate, allowing for more complex possibilities.

**Comment by user2684** [comment]
Hi I think the API described herehttps://open.imoulife.com/book/en/http/device/operate/controlLocationPTZ.html seems the one which allows moving the camera to a given position. Let me add it to the development queue even if the implementation could take a while :-) Thanks

**Comment by bimbopp** [comment]
Hi would it still be possible to add the devicePTZInfo service?

**Comment by user2684** [comment]
You mean having a service which moves to a preset position? Thanks

**Comment by Brix1330** [comment]
> You mean having a service which moves to a preset position? Thanks

Yes, and it would also be even better to have a GUI control (device info page) as well. Thanks.

**Comment by mguyard** [comment]
Hi,

Also very interrested by this feature

**Comment by mguyard** [comment]
Hello,

With Imou mobile app we are able to set favorite (preset). That seems to be available in api :

- Retrieve favorite : https://openapi-[datacenter].easy4ip.com/openapi/getCollection
- Set Favorite : https://openapi-[datacenter].easy4ip.com/openapi/setCollection
- Modify Favorite : https://openapi-[datacenter].easy4ip.com/openapi/modifyCollection
- Delete Favorite : https://openapi-[datacenter].easy4ip.com/openapi/deleteCollection
- Move to a Favorite : https://openapi-[datacenter].easy4ip.com/openapi/turnCollection

With these API endpoint, preset will be easy to manage. Do you think you can implement this ?

**Comment by user2684** [comment]
Hi, try to have a look at [this](https://github.com/user2684/imou_life/wiki/Is-this-an-official-integration%3F), I think the functionality is already there

**Comment by user2684** [comment]
Hi, closing since I have recently decided to stop actively developing the integration and add new capabilities. For more information, have a look at https://github.com/user2684/imou_life/wiki/Status-of-the-integration
